update-version-and-changelog.js

```diff
@@ -0,0 +1,84 @@
+const semver = require('semver');
+const fs = require('fs');
+const core = require('@actions/core');
+const simpleGit = require('simple-git');
+
+const git = simpleGit.default();
+
+const releaseType = process.env.RELEASE_TYPE;
+const VALID_RELEASE_TYPES = ['major', 'minor', 'patch'];
+
+// To get the merges since the last tag
+async function getChangesSinceGitTag (tag) {
+    const changes = await git.log(["--reverse", "--merges", `HEAD...${tag}`]);
+    return changes;
+}
+
+// To know if there are changes since the last tag.
+// we are not using getChangesSinceGitTag because it returns the just the merges and not the commits.
+// So for example if a hotfix was committed directly to trunk this function will detect it but getChangesSinceGitTag will not.
+async function getHasChangesSinceGitTag (tag) {
+    const changes = await git.log([`HEAD...${tag}`]);
+    return changes?.all?.length > 0;
+}
+
+async function updateVersion () {
+    if ( !VALID_RELEASE_TYPES.includes(releaseType) ) {
+        console.error("❎  Error: Release type is not valid. Valid release types are: major, minor, patch.");
+        process.exit(1);
+    }
+
+    if (!fs.existsSync('./package.json')) {
+        console.error("❎  Error: package.json file not found.");
+        process.exit(1);
+    }
+
+    if (!fs.existsSync('./readme.txt')) {
+        console.error("❎  Error: readme.txt file not found.");
+        process.exit(1);
+    }
+
+    const package = require('./package.json');
+    const currentVersion = package.version;
+    const newVersion = semver.inc(currentVersion, releaseType);
+    const currentTag = `v${currentVersion}`;
+    const newTag = `v${newVersion}`;
+
+    if (!semver.valid(currentTag)) {
+        console.error(`❎  Error: current tag ( ${ currentTag } ) is not a valid semver version."`);
+        process.exit(1);
+    }
+
+    // get changes since last tag
+    const changes = await getChangesSinceGitTag(currentTag);
+    const hasChangesSinceGitTag = await getHasChangesSinceGitTag(currentTag);
+
+    // check if there are any changes
+    if (!hasChangesSinceGitTag) {
+        console.error("❎  No changes since last tag. There is nothing to release.");
+        process.exit(1);
+    }
+
+    // update package.json version
+    package.version = newVersion;
+    fs.writeFileSync('./package.json', JSON.stringify(package, null, 2));
+    console.info('✅ Version updated', currentTag, '=>', newTag);
+    
+    // update readme.txt version with the new changelog
+    const readme = fs.readFileSync('./readme.txt', 'utf8');
+    const changelogChanges = changes.all.map(change => `* ${ change.body || change.message }`).join('\n');
+    const newChangelog = `== Changelog ==\n\n= ${ newVersion } =\n\n${ changelogChanges }`;
```

Comment:
Should it be two line breaks or one, between the `newVersion` and the changes? 😄 
```suggestion
    const newChangelog = `== Changelog ==\n\n= ${ newVersion } =\n${ changelogChanges }`;
```

Reply:
In the current version, we have only one but we can switch to 2 :)